tools/create_db.py

```diff
@@ -297,6 +331,121 @@ def create_db(input_file, output_dir,
 
     logger.info('Database created after %d seconds.' % (time.time() - start))
 
+def read_list(images_list_mxnet):
+    image_list = []
+    with open(images_list_mxnet) as fin:
+        for line in fin.readlines():
+            line = [i.strip() for i in line.strip().split('\t')]
+            item = [int(line[0])] + [line[-1]] + [float(i) for i in line[1:-1]]
+            image_list.append(item)
+    return image_list
+	
+def _create_rec(image_count,images_folder, images_list_mx,image_height,image_width,
+               resize_mode,image_channels,encoding,output_dir,prefix,num_threads,mean_files):
+	source = images_list_mx
+    tic = [time.time()]
+    total = len(source)
+	
+    def image_encode(item, q_out):
+        try:
+		    img = cv2.imread(item[1])
+        except:
+            print 'imread error:', item[1]
+            return
+        if img == None:
+            print 'read none error:', item[1]
+            return
+		img = utils.image.image_to_array(img, image_channels)
+		
+		img = utils.image.resize_image(img,
+                image_height, image_width,
+                channels    = image_channels,
+                resize_mode = resize_mode,
+                )
+			
+        header = mx.recordio.IRHeader(0, item[2], item[0], 0)
+        try:
+            s = mx.recordio.pack_img(header, img, quality=90, img_fmt=encoding)
+            q_out.put(('data', s, item))
+        except:
+            print 'pack_img error:',item[1]
+            return
+
+    def read_worker(q_in, q_out):
+        while not q_in.empty():
+            item = q_in.get()
+            image_encode(item, q_out)
+
+    def write_worker(q_out, prefix):
+        pre_time = time.time()
+        sink = []
+        record = mx.recordio.MXRecordIO(prefix+'.rec', 'w')
+        while True:
+            stat, s, item = q_out.get()
+            if stat == 'finish':
+                write_list(prefix+'.lst', sink)
+                break
+            record.write(s)
+            sink.append(item)
+            if len(sink) % 1000 == 0:
+                cur_time = time.time()
+                print 'time:', cur_time - pre_time, ' count:', len(sink)
+                pre_time = cur_time
+
+    try:
+        import multiprocessing
```

Comment:
@lukeyeager Thank you. I imitate the source of the mxnet. But your advice is good.